sdk/core/core-http/lib/util/utils.ts

```diff
@@ -241,3 +241,17 @@ export function replaceAll(
 export function isPrimitiveType(value: any): boolean {
   return (typeof value !== "object" && typeof value !== "function") || value === null;
 }
+
+/**
+ * @ignore
+ * Helper utility to clean up unintended characters that get appended by OS.
```

Comment:
My understanding is that the OS won't automatically append the BOM to a file as it doesn't know the content of the file. It is up to the application to add those bytes or not.

In our scenario, who is going to put the BOM into the stream that contains the XML doc?

Reply:
This is the comment that was added by original authors in `azure-sb`

Based on https://www.w3.org/International/questions/qa-byte-order-mark.en, looks like this gets added commonly by editors and any app that's used to generate the text content.

Reply:
I understand the scenario where applications put this header. What I am trying to understand is how is the service bus SDK receiving XML objects that contain this BOM. 

I don't mind adding this in if there is a scenario in which we know the BOM is being returned by the service -- what I am asking is for is that scenario.

Reply:
So there is no test scenario that was written earlier that surfaced this as being sent by the service.
As discussed earlier, our understanding is that the OS on which the code generating files (if any) would end up appending this.

So no, I haven't encountered a concrete scenario supporting this other than the fact that this check was added with a comment describing reason by authors in `azure-sb`

Notes from offline discussion yesterday afternoon for ref -
_•	Handling of unwanted BOM characters - This was being taken care of in azure-sb implementation and seems less understood by us overall. 
https://en.wikipedia.org/wiki/Byte_order_mark
https://www.screenaware.com/en/blog/xml2js-sax-js-non-whitespace-before-first-tag
Based on research and testing things, looks like it's a general problem that users face and must have been added for a reason in azure-sb. The documentation and intent is pretty clear too. 
In Node, while xml2js handles this internally for XML parsing, it doesn't do so for XML building. 
In browser, both building and parsing require BOM handling._